CMakeLists.txt

```
@@ -68,6 +68,25 @@ option( TRACCC_USE_SYSTEM_LIBS "Use system libraries be default" FALSE )
 option( TRACCC_USE_SPACK_LIBS "Use pack libraries by default" FALSE )
 option( TRACCC_USE_ROOT "Use ROOT in the build (if needed)" TRUE )
 
+# Check for some compiler features
+if(TRACCC_BUILD_EXAMPLES)
+   message(STATUS "Checking for `std::format` support")
+   file( WRITE
+   "${CMAKE_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/CMakeTmp/compile_test_format.cpp"
+   "#include <format>\n"
+   "int main() {\n"
+   "std::format(\"Hello {}\\n\", \"world\");\n"
+   "return 0; }\n" )
+   try_compile( TRACCCC_INTERNAL_STD_FORMAT_WORKS SOURCES
+   "${CMAKE_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/CMakeTmp/compile_test_format.cpp")
+
+   if (NOT TRACCCC_INTERNAL_STD_FORMAT_WORKS)
+      message(FATAL_ERROR "Examples are enabled, but `std::format` is not supported by the C++ compiler; upgrade to gcc 13 or clang 14 or higher.")
+   endif()
+
+   unset(TRACCCC_INTERNAL_STD_FORMAT_WORKS)
```

No need to write such low level code. We have [check_cxx_source_compiles(...)](https://cmake.org/cmake/help/latest/module/CheckCXXSourceCompiles.html) for this. Which is used extensively in [vecmem](https://github.com/acts-project/vecmem) for instance. (https://github.com/acts-project/vecmem/blob/main/core/CMakeLists.txt#L208)

I'd propose:

```suggestion
   include(CheckCxxSourceCompiles)
   check_cxx_source_compiles( "
      #include <format>
      int main() {
         std::format(\"Hello {}\\n\", \"world\");
      }
      " TRACCC_STD_FORMAT_WORKS )
   if( NOT TRACCC_STD_FORMAT_WORKS )
      message(FATAL_ERROR "Examples are enabled, but `std::format` is not supported by the C++ compiler; upgrade to gcc 13 or clang 14 or higher.")
   endif()
```